Doc/library/math.rst

```diff
@@ -22,8 +22,8 @@ instead of a complex result allows earlier detection of the unexpected complex
 number used as a parameter, so that the programmer can determine how and why it
 was generated in the first place.
 
-The following functions are provided by this module.  Except when explicitly
-noted otherwise, all return values are floats.
+The following functions are provided by this module.  Return values for module
+functions are either floats, or determined from acceptable types of arguments.
```

Comment:
To me this reads a bit like: the return value is float, or something else. Which is more or less saying: the return value is something.

Also there are methods like `math.isfinite` that always return a `bool`, independent of the input type. 

And finally there are cases where the return type depends on the value, and not only on the type of the input arguments:
```
import math
import numpy as np

type((math.sumprod([10**20], [np.int8(3)]))) # type int
type((math.sumprod([10], [np.int8(3)]))) # type np.int32
```

I seems hard to state exactly what the return type will be, so maybe formulate a bit broad?

> Except when explicitly noted otherwise, all return values are floats or determined from the input arguments

Reply:
> To me this reads a bit like: the return value is float, or something else. Which is more or less saying: the return value is something.

Yeah, I think I should restore leading "Except when explicitly noted otherwise".

> Also there are methods like math.isfinite that always return a bool, independent of the input type.

This is explicitly noted:)

> And finally there are cases where the return type depends on the value, and not only on the type of the input arguments

I'm not sure.  It depends on coercion rules for arguments and hard to describe in details.  For example, it seems you are on numpy-1.x; I've instead (with CPython 3.12, BTW) got:
```pycon
>>> np.__version__
'2.1.0'
>>> math.sumprod([10], [np.int8(3)])
np.int8(30)
>>> math.sumprod([10**20], [np.int8(3)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python int too large to convert to C long
```

So, I think, that fits to the scheme.  This is vague, but give some hints on what's going on.

> maybe formulate a bit broad?

I'm open to suggestions.  Original proposal (Raymond's) is on the table too: we can modify instead docs for prod/sumprod (maybe more?), where mention that "the output type is determined by the input type".